data_model/src/asset.rs

```diff
@@ -519,7 +524,7 @@ impl HasMetadata for NewAssetDefinition {
 #[cfg_attr(feature = "ffi_import", iroha_ffi::ffi_import)]
 impl NewAssetDefinition {
     /// Create a [`NewAssetDefinition`], reserved for internal use.
-    fn new(id: <AssetDefinition as Identifiable>::Id, value_type: AssetValueType) -> Self {
+    pub fn new(id: <AssetDefinition as Identifiable>::Id, value_type: AssetValueType) -> Self {
```

Comment:
this shouldn't be a public fn. You have at your disposal `AssetDefinition::quantity`, `AssetDefinition::store`, etc

Reply:
Why not? It is not possible to create any kind of inconsistent asset definition using `NewAssetDefinition::new`.

I made this function public because i want to create different types of asset definitions inside loop.
I can use function pointers instead, but it sounds a bit more complicated.

Reply:
it's a matter of consistent API. We don't expose any methods on builders. You can add this function to `AssetDefinition` and use `AssetDefinition::new(definition_id.clone(), asset_value_type)`:
```rs
    pub fn new(id: <Self as Identifiable>::Id, value_type: AssetValueType) -> <Self as Registered>::With {
        <Self as Registered>::With::new(id, value_type)
    }
```